### PR TITLE
Add loadBalancer client config.

### LINKED ISSUE #147: configurable load balancing strategies

### PR DESCRIPTION
**Pull request by adleong**
Fixes #147 

Add a loadBalancer section to the client config where a load balancer can be
specified and configured.  The load balancers that are currently supported are
p2c, ewma, aperture, and heap.

### CODE REVIEW

File: linkerd/core/src/test/scala/io/buoyant/linkerd/LoadBalancerTest.scala [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      |  client:
+      |    loadBalancer:
+      |      kind: $balancer
+      |  servers:
+      |  - {}
+      |""".stripMargin
+
+      val linker = Linker.load(config, Seq(TestProtocol.Plain))
+      val factory = linker.routers.head.params[LoadBalancerFactory.Param]
+      assert(factory.loadBalancerFactory != DefaultBalancerFactory)
```
Comment (adleong) [inline linkerd/core/src/test/scala/io/buoyant/linkerd/LoadBalancerTest.scala]: I wish we could test more here, but finagle is pretty opaque about the load balancer.

File: docs/config.md [hunk 2]
```diff
@@ -197,6 +201,47 @@ names:
   commonNamePattern: "{host}.buoyant.io"
 ```
 
+#### Load Balancer
+
```
Comment (gtcampbell) [inline docs/config.md]: it might be good to note here what default load balancer is used if this option is unspecified (unless I missed that?)

**Comment by gtcampbell** [comment]
:star: This looks like a useful thing to be able to configure.

**Comment by siggy** [comment]
lgtm :+1: :star:

File: docs/config.md (hunk 2 above)
Comment (adleong) [inline docs/config.md]: Good point.  The default (p2c) comes from finagle so I'm a bit uneasy documenting something that could change out from under us.  But in this case I think it makes sense.

Comment (olix0r) [inline docs/config.md]: also, per @stevej, i think we'd _like_ to change the default for routers to be EWMA...

**Comment by adleong** [comment]
:star: rebase

File: docs/config.md [hunk 3]
```diff
@@ -197,6 +201,47 @@ names:
...
+* *maxEffort* -- Optional.
+
+##### [ewma][ewma]
+
+ewma supports the following options (see [here][ewma] for option semantics and defaults):
+
+* *maxEffort* -- Optional.
+* *decayTimeMs* -- Optional.
+
```
Comment (olix0r) [inline docs/config.md]: per my understanding, p2c and ewma (and apperature) aren't mutually exclusive. I believe that EWMA is a costing strategy to feed into P2C, and apperature is a windowing strategy taht still uses P2C.  I think that "p2c" should actually be "queuedepth", here -- but I defer to someone who knows more about the details...

Comment (adleong) [inline docs/config.md]: That's right.  What we call ewma here is actually p2c+ewma.  Similarly, what we call aperture is actually aperture+least loaded.  The finagle docs (which we link to) spell this out more explicitly.

**Pull request merged**